storage/remote/queue_manager.go

```diff
@@ -891,6 +1007,20 @@ func (s *shards) sendSamplesWithBackoff(ctx context.Context, samples []prompb.Ti
 	}
 }
 
+func buildMetadataWriteRequest(metadata []*prompb.MetricMetadata) ([]byte, error) {
```

Comment:
I think `buildMetadataWriteRequest` and `buildWriteRequest` should be merged - add a `metadata` parameter to that `buildWriteRequest`.